org.idempiere.test/src/org/idempiere/test/model/MSequenceTest.java

```diff
@@ -0,0 +1,137 @@
+    private static final String TEST_SUFFIX = "-End";
+    private static final String TEST_DATE = "2025-06-13 10:00:00";
+    private static final String TEST_DATE_yyyyMM = "202506";
+
+    @Test
+    public void testDocumentNoWithContextVariableKey() {
+        Properties ctx = Env.getCtx();
+        String trxName = getTrxName();
+
+        // Create test sequence
+        MSequence seq = new MSequence(ctx, 0, trxName);
+        seq.setName("Test Sequence with Context Key");
+        seq.setDescription("Test sequence for document numbers with context variable key");
+        seq.setPrefix(TEST_PREFIX + "@Updated<yyyyMM>/K@"); // /K flag indicates context variable use
+        seq.setSuffix("@AD_Org_ID/K@" + TEST_SUFFIX);
+        seq.setStartNo(1000);
+        seq.setCurrentNext(1000);
+        seq.setIncrementNo(1);
+        seq.setIsActive(true);
+        seq.setIsAutoSequence(true);
+        seq.setStartNewYear(false);
+        seq.setStartNewMonth(false);
+        seq.saveEx();
+
+        // Create test document type
+        MDocType docType = new MDocType(ctx, 0, trxName);
+        docType.setName("Test Document Type");
+        docType.setPrintName("Test Document");
+        docType.setDocNoSequence_ID(seq.get_ID());
+        docType.setIsActive(true);
+        docType.setDocBaseType("SOO"); // Sales Order
+        docType.setGL_Category_ID(); // Set default GL category
+        docType.setIsDocNoControlled(true);
+        docType.saveEx();
+
+        try (MockedStatic<MDocType> mockedStatic = Mockito.mockStatic(MDocType.class)) {
+        	mockedStatic.when(() -> MDocType.get(ctx, docType.getC_DocType_ID()))
+				.thenReturn(docType);
+	        // Generate document number through sales order creation
+	        Timestamp testDate = Timestamp.valueOf(TEST_DATE);
+	        Env.setContext(ctx, "#Date", testDate);
+	        int testOrgID = 11; // Example organization ID
+	        Env.setContext(ctx, "#AD_Org_ID", testOrgID);
+
+	        // Create and save a sales order
+	        MOrder order = new MOrder(ctx, 0, trxName);
+	        order.setBPartner(MBPartner.get(ctx, DictionaryIDs.C_BPartner.JOE_BLOCK.id));
+	        order.setC_DocTypeTarget_ID(docType.getC_DocType_ID());
+	        order.setDeliveryRule(MOrder.DELIVERYRULE_CompleteOrder);
+	        order.setDocStatus(DocAction.STATUS_Drafted);
+	        order.setDocAction(DocAction.ACTION_Complete);
+	        order.setDateOrdered(testDate);
+	        order.setDatePromised(testDate);
+	        order.saveEx();
+
+	        String docNo = order.getDocumentNo();
+	        assertNotNull(docNo, "Document number should not be null");
+
+	        // The document number should follow pattern: TestSeq-202506{seq}11-End
+	        assertTrue(docNo.startsWith(TEST_PREFIX + TEST_DATE_yyyyMM),
+	                "Document number should start with prefix and date key");
+	        assertTrue(docNo.endsWith(testOrgID + TEST_SUFFIX),
+	                "Document number should end with org ID key and suffix");
```

Comment:
Suggest to change it to "Document number should end with org ID key and suffix: "+docNo